### DISCUSSION TITLE
Strimzi kafka with Istio gateway in Peerauthentication Permissive mode shows error in kafka for ssl handshake on 9091

### DISCUSSION THREAD

**Discussion by seemasanjaisinghani** [strimzi/strimzi-kafka-operator#6175] [Q&A]
Hi Team,

We are integrating Strimzi Kafka with Istio gateway for service discovery in our organization.

Here is the Kafka Cluster yaml we are using for installation of kafka 👍  ->
--------------------------------------------------------->
`apiVersion: kafka.strimzi.io/v1beta2
kind: Kafka
metadata:
  name: {{ .Values.kafka_cluster_name }}
spec:
  clusterCa:
    generateCertificateAuthority: {{ .Values.kafka_generate_cert }}
  clientsCa:
    generateCertificateAuthority: {{ .Values.kafka_generate_client_cert }}
  kafka:
    version: 2.8.0
    replicas: {{ .Values.kafka_replicas }}
    logging:
      type: inline
      loggers:
        kafka.authorizer.logger: "DEBUG"
    listeners:
      - name: plain
        port: 9092
        type: internal
        tls: false
      - name: external
        port: 9094
        type: loadbalancer
        tls: false
        configuration:
          loadBalancerSourceRanges:
            - {{ .Values.kafka_listeners_external_load_balancer_source_ranges }}
          bootstrap:
            annotations:
              cloud.google.com/load-balancer-type: "Internal"
          brokers:
          - broker: 0
            annotations:
              cloud.google.com/load-balancer-type: "Internal"
          - broker: 1
            annotations:
              cloud.google.com/load-balancer-type: "Internal"
          - broker: 2
            annotations:
              cloud.google.com/load-balancer-type: "Internal"
    config:
      offsets.topic.replication.factor: {{ .Values.kafka_offsets_topic_replication_factor }}
      transaction.state.log.replication.factor: {{ .Values.kafka_transaction_state_log_replication_factor }}
      transaction.state.log.min.isr: 2
      log.message.format.version: "2.8"
      inter.broker.protocol.version: "2.8"
      security.inter.broker.protocol: "PLAINTEXT"
      zookeeper.connection.timeout.ms: 18000
      ssl.endpoint.identification.algorithm: ""
      listener.name.internal.ssl.endpoint.identification.algorithm: ""
    template:
      pod:
        metadata:
          annotations:
            sidecar.istio.io/inject: "true"
            traffic.sidecar.istio.io/includeInboundPorts: "*"
            traffic.sidecar.istio.io/excludeInboundPorts: "2888,3888,9091,9090,9092,9093,9094"
            traffic.sidecar.istio.io/excludeOutboundPorts: "2888,3888,9091,9090,9092,9093,9094"
            #traffic.sidecar.istio.io/excludeOutboundPorts: "9091,9090,2181,2888,3888"
          labels:
          {{- if .Values.istiolabel }}
            istio.io/rev:  {{ .Values.istiolabel }}
          {{- end }}
            
    storage:
      type: {{ .Values.kafka_storage_type }}
      volumes:
      - id: 0
        type: persistent-claim
        size: {{ .Values.kafka_storage_volume_size }}
        deleteClaim: true
  zookeeper:
    replicas: {{ .Values.zookeeper_replicas }}  
    config:
      quorumListenOnAllIPs: true
    storage:
      type: persistent-claim
      size: {{ .Values.zookeeper_storage_size }}
      deleteClaim: true
    template:
      pod:
        metadata:
          annotations:
            sidecar.istio.io/inject: "true"
            traffic.sidecar.istio.io/includeInboundPorts: "*"
            traffic.sidecar.istio.io/excludeInboundPorts: "2888,3888,9091,9090,9094,9093"
            traffic.sidecar.istio.io/excludeOutboundPorts: "2888,3888,9091,9090,9094,9093"
          labels:
          {{- if .Values.istiolabel }}
            istio.io/rev:  {{ .Values.istiolabel }}
          {{- end }}
  entityOperator:
    topicOperator: {}
    userOperator: {}
    template:
      pod:
        metadata:
          annotations:
            sidecar.istio.io/inject: "true"
            traffic.sidecar.istio.io/includeInboundPorts: "*"
            traffic.sidecar.istio.io/excludeInboundPorts: "2888,3888,9091,9090,9092,9093,9094"
            traffic.sidecar.istio.io/excludeOutboundPorts: "2888,3888,9091,9090,9092,9093,9094"
          labels:
          {{- if .Values.istiolabel }}
            istio.io/rev:  {{ .Values.istiolabel }}
          {{- end }}`
<----------------------------------------------------------------

When used the PeerAuthentication mtls mode : DISABLE, the kafka cluster comes up fine.
peer.yaml->
================================
apiVersion: "security.istio.io/v1beta1"
kind: "PeerAuthentication"
metadata:
  name: "peerauth-orch"
  namespace: "awg-kafka"
spec:
  mtls:
    mode: DISABLE
================================

But with PeerAuthentication mtls mode : PERMISSIVE , Kafka shows below error - 
INFO [SocketServer listenerType=ZK_BROKER, nodeId=2] Failed authentication with /240.0.210.6 (SSL handshake failed) (org.apache.kafka.common.network.Selector) [data-plane-kafka-network-thread-2-ListenerName(REPLICATION-9091)-SSL-5]

================================
peer.yaml->
apiVersion: "security.istio.io/v1beta1"
kind: "PeerAuthentication"
metadata:
  name: "peerauth-orch"
  namespace: "awg-kafka"
spec:
  mtls:
    mode: PERMISSIVE
================================

Any help would be appreciated, as we are stuck with this issue.

**Comment by seemasanjaisinghani** [comment]
@scholzj Any help?

**Reply by scholzj** [reply]
Strimzi has currently no official support for Istio and I do not know how to run it with Istio. So I'm afraid can't help with this.

**Comment by seemasanjaisinghani** [comment]
@scholzj Thank you!
One more quick question , if I want to set quorumListenOnAllIPs=true for zookeeper in kafka.yaml, how I can do that?

**Reply by scholzj** [reply]
I don't really know, I never needed to change anything about that option. Have you tried to add it to the config section?

**Reply by seemasanjaisinghani** [reply]
@scholzj I have tried this below way, but not sure if that's correct way?

  zookeeper:
    replicas: {{ .Values.zookeeper_replicas }}  
    config:
      quorumListenOnAllIPs: true
    storage:
      type: persistent-claim
      size: {{ .Values.zookeeper_storage_size }}
      deleteClaim: true
    template:
      pod:
        metadata:
          annotations:
            sidecar.istio.io/inject: "true"
            traffic.sidecar.istio.io/includeInboundPorts: "*"
            traffic.sidecar.istio.io/excludeInboundPorts: "2888,3888,9091,9090,9094,9093"
            traffic.sidecar.istio.io/excludeOutboundPorts: "2888,3888,9091,9090,9094,9093"

**Reply by scholzj** [reply]
If adding it to tje `.spec.zookeeper.config` does not work than you probably cannot set it.

**Reply by scholzj** [reply]
But from the code, it does not seem to be forbidden or blocked by Strimzi in any way. SO not sure why would it not work.
